### PR TITLE
toString added to MultiPartIDJavaRefinement

### PR DESCRIPTION
**Pull request by olhabn**

### CODE REVIEW

File: src/foam/java/refinements.js [hunk 1]
```diff
@@ -2116,6 +2116,18 @@ foam.CLASS({
 
         return str;
       }
+    },
+    {
+      name: 'toString',
```
Comment (olhabn) [inline src/foam/java/refinements.js:2121]: this doesn't work I can try to investigate it later

File: src/foam/core/IDSupport.js [hunk 2]
```diff
@@ -133,8 +140,17 @@ foam.CLASS({
...
         methods: [
-          function toString() {
-            return foam.json.Compact.stringify(this, this.cls_);
+          {
+            name: 'toString',
+            code: function() {
+              console.log(this);
+              var arr = [];
+              for ( var prop in this.instance_ ) {
+                arr.push(! this[prop] ? '' : this[prop].toSummary ? this[prop].toSummary() : this[prop].toString() );
```
Comment (kgrgreer) [inline src/foam/core/IDSupport.js:149]: Why toSummary?

Comment (olhabn) [reply src/foam/core/IDSupport.js:149]: in case on of the keys is an object

Comment (kgrgreer) [reply src/foam/core/IDSupport.js:149]: Can objects be keys?

File: src/foam/core/IDSupport.js [hunk 3]
```diff
@@ -133,8 +140,17 @@ foam.CLASS({
           return prop.clone();
         }),
         methods: [
-          function toString() {
-            return foam.json.Compact.stringify(this, this.cls_);
+          {
+            name: 'toString',
+            code: function() {
+              console.log(this);
```
Comment (kgrgreer) [inline src/foam/core/IDSupport.js:146]: remove

File: src/foam/core/IDSupport.js [hunk 4]
```diff
@@ -133,8 +140,17 @@ foam.CLASS({
...
+              var arr = [];
+              for ( var prop in this.instance_ ) {
+                arr.push( ! this[prop] ? '' : this[prop].toString() );
```
Comment (kgrgreer) [inline src/foam/core/IDSupport.js:149]: When possible, better to remove the '!' and then reverse the two sides of the '?'.

File: src/foam/core/IDSupport.js (hunk 3 above)
Comment (olhabn) [reply src/foam/core/IDSupport.js]: fixed

File: src/foam/core/IDSupport.js (hunk 4 above)
Comment (olhabn) [reply src/foam/core/IDSupport.js]: fixed

**Pull request merged**